internal/schema/schema.go

```diff
@@ -48,3 +48,17 @@ func SetExistingAttributesAsOptional(schema map[string]*schema.Schema, keys ...s
 		schema[attribute].Required = false
 	}
 }
+
+// Clone returns a shallow clone of m.
```

Comment:
For our use case a shallow clone is perfectly fine as the attributes are top-level.